Mathlib/Analysis/Convex/StrictConvexBetween.lean

```diff
@@ -96,6 +96,27 @@ lemma dist_add_dist_eq_iff : dist a b + dist b c = dist a c ↔ Wbtw ℝ a b c :
     at this
   rwa [(vsub_left_injective _).mem_set_image] at this
 
+/-- The strict triangle inequality. -/
+theorem dist_lt_dist_add_dist_iff {a b c : P} :
+    dist a c < dist a b + dist b c ↔ ¬ Wbtw ℝ a b c := by
+  rw [← ne_iff_lt_iff_le.mpr (dist_triangle _ _ _), not_iff_not, eq_comm, dist_add_dist_eq_iff]
+
+/-- The strict triangle inequality for affinely independent points. -/
+theorem AffineIndependent.dist_strict_triangle {a b c : P} (hT : AffineIndependent ℝ ![a, b, c]) :
+    dist a c < dist a b + dist b c := by
+  rw [dist_lt_dist_add_dist_iff]
+  rw [affineIndependent_iff_not_collinear_set] at hT
+  exact mt (·.collinear) hT
+
+/-- Alternate form of the strict triangle inequality, allowing an ordered triple of
+points to be specified from a possibly larger collection `T` of affinely independent points. -/
+theorem AffineIndependent.dist_strict_triangle' {ι} (i j k : ι)
+    (h : Function.Injective ![i, j, k]) (T : ι → P) (hT : AffineIndependent ℝ T) :
+    dist (T i) (T k) < dist (T i) (T j) + dist (T j) (T k) := by
+  refine AffineIndependent.dist_strict_triangle ?_
+  convert hT.comp_embedding ⟨_, h⟩ using 1
+  exact FinVec.map_eq _ ![i, j, k]
```

Comment:
It's not clear to me whether either of these are good lemmas any more

Reply:
They seem useful to me: https://github.com/dwrensha/compfiles/blob/de3893c2b96a6352d03db561c52eba51ce7731da/Compfiles/Imo1983P6.lean#L164-L173

@eric-wieser do you have any suggestions on how to make them better?

Reply:
Even with the new `dist_lt_dist_add_dist_iff` above?

Reply:
Ah, I had not seen that.
It still seems to require me to do some awkward juggling of indices. How would you finish this proof?

```lean
example (T : Affine.Triangle ℝ (EuclideanSpace ℝ (Fin 2))) :
    dist (T.points 1) (T.points 0) <
      dist (T.points 1) (T.points 2) + dist (T.points 2) (T.points 0) := by
  rw [dist_lt_dist_add_dist_iff]
  -- ... ?
  sorry

```

Reply:
Maybe some API around `AffineIndependent` and `Not (Btw _ _ _)`, perhaps similar to what we wrote here?

Reply:
@eric-wieser do you mean like this?
```lean
theorem AffineIndependent.not_sbtw {a b c : P} (hT : AffineIndependent ℝ ![a, b, c]) :
    ¬ Wbtw ℝ a b c := by
  rw [affineIndependent_iff_not_collinear_set] at hT
  exact mt (·.collinear) hT

theorem AffineIndependent.not_sbtw' {ι} (i j k : ι)
    (h : Function.Injective ![i, j, k]) (T : ι → P) (hT : AffineIndependent ℝ T) :
    ¬ Wbtw ℝ (T i) (T j) (T k):= by
  refine AffineIndependent.not_sbtw ?_
  convert hT.comp_embedding ⟨_, h⟩ using 1
  exact FinVec.map_eq _ ![i, j, k]
```

That would be sufficient for my needs.

I'm not sure what file these lemmas would live in, though.

Reply:
Yes, those look reasonable enough (though that should say `wbtw` not `sbtw`)

Reply:
Fixed the naming and generalized the ring in `https://github.com/leanprover-community/mathlib4/pull/24104/commits/a8ccb448286d798aef6c8e9594298a033500282e`.